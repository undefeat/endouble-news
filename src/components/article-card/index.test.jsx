import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import shallow from 'react-test-renderer/shallow';
import ArticleCard from '.';

const article = {
    source: 'Ars Technica',
    author: 'Sean Gallagher',
    title: 'Google releases Chrome extension that alerts to breached passwords - Ars Technica',
    description: 'Using hashed and encrypted store, add-on securely checks logins against breach database.',
    url:
        'https://arstechnica.com/information-technology/2019/02/google-releases-chrome-extension-that-alerts-to-breached-passwords/',
    urlToImage: 'https://cdn.arstechnica.net/wp-content/uploads/2019/02/passwordcheck-760x380.jpg',
    publishedAt: '2019-02-05T14:54:00Z',
    content:
        "Enlarge / Good news. For now. 23 with 21 posters participating With lists of billions of compromised credentials floating around on underground forums and in text-paste pages across the Internet, it's difficult for anyone to keep up with the potential threat â€¦ [+2566 chars]",
};

describe('ArticleCard', () => {
    it('should correctly render', () => {
        const tree = shallow.createRenderer().render(<ArticleCard article={article} />);

        expect(tree).toMatchSnapshot();
    });

    it('should change "modalIsOpen" when buttons are clicked', () => {
        const element = ReactTestUtils.renderIntoDocument(<ArticleCard article={article} />);

        expect(element.state.modalIsOpen).toBe(false);
        const openButton = ReactTestUtils.findRenderedDOMComponentWithClass(element, 'article-card__modal-btn');
        ReactTestUtils.Simulate.click(openButton);
        expect(element.state.modalIsOpen).toBe(true);

        const closeButton = ReactTestUtils.findRenderedDOMComponentWithClass(element, 'modal__close-btn');
        ReactTestUtils.Simulate.click(closeButton);
        expect(element.state.modalIsOpen).toBe(false);
    });
});
