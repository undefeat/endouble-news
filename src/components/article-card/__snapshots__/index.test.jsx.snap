// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard should correctly render 1`] = `
<li
  className="article-card"
>
  <article>
    <button
      aria-label="Open in modal"
      className="article-card__modal-btn"
      onClick={[Function]}
      type="button"
    >
      <figure
        className="article-card__thumbnail"
      >
        <img
          alt="thumbnail"
          className="article-card__thumbnail__img"
          src="https://cdn.arstechnica.net/wp-content/uploads/2019/02/passwordcheck-760x380.jpg"
        />
      </figure>
      <h2
        className="article-card__title"
      >
        Google releases Chrome extension that alerts to breached passwords - Ars Technica
      </h2>
    </button>
    <ArticleMeta
      publishedAt="2019-02-05T14:54:00Z"
      source="Ars Technica"
      url="https://arstechnica.com/information-technology/2019/02/google-releases-chrome-extension-that-alerts-to-breached-passwords/"
    />
    <p
      className="article-card__description"
    >
      Using hashed and encrypted store, add-on securely checks logins against breach database.
    </p>
  </article>
</li>
`;
